subprojects/docs/src/docs/userguide/workingWithFiles.adoc

```diff
@@ -135,90 +461,224 @@ Usually, there is a method with the same name as the property, which appends to
+
+----
+def webAssetsSpec = copySpec {
+    include '**/*.html', '**/*.png', '**/*.jpg'
+}
+
+task copyAssets(type: Copy) {
+    into "${buildDir}/explodedWar"
+    with webAssetsSpec
+}
+
+task archiveApp(type: Zip) {
+    from "otherResources"
+    with webAssetsSpec
+}
+----
+
+As you can see, the api:org.gradle.api.file.CopySpec#with(org.gradle.api.file.CopySpec...)[] method is the key to incorporating an existing copy specification into another one.
+
+===== Using child specifications
+
+If you only use a single copy spec, the file filtering and renaming will apply to _all_ the files that are copied. Sometimes this is what you want, but not always. Consider the following example that copies files into a directory structure that can be used by a Java Servlet container to deliver a website:
+
+++++
+<figure>
+    <title>Creating an exploded WAR for a Servlet container</title>
+    <imageobject>
+        <imagedata fileref="img/exploded-war-child-copy-spec-example.png" width="160mm"/>
+    </imageobject>
+</figure>
+++++
+
+This is not a straightforward copy as the `WEB-INF` directory and its subdirectories don't exist within the project, so they must be created during the copy. In addition, we only want HTML and image files going directly into the root folder — `build/explodedWar` — and only JavaScript files going into the `js` directory. So we need separate filter patterns for those two sets of files.
+
+The solution is to use _child specifications_, which can be applied to both `from()` and `into()` declarations. The following task definition does the necessary work:
 
 ++++
 <sample id="nestedCopySpecs" dir="userguide/files/copy" title="Nested copy specs">
     <sourcefile file="build.gradle" snippet="nested-specs"/>
 </sample>
 ++++
 
+Notice how the `src/dist` configuration has a nested inclusion specification: that's the child copy spec. You can of course add content filtering and renaming here as required. A child copy spec is still a copy spec.
+
+The above example also demonstrates how you can copy files into a subdirectory of the destination either by using a child `into()` on a `from()` or a child `from()` on an `into()`. Both approaches are acceptable, but you may want to create and follow a convention to ensure consistency across your build files.
+
+[NOTE]
+Don't get your `into()` specifications mixed up! For a normal copy — one to the filesystem rather than an archive — there should always be _one_ "root" `into()` that simply specifies the overall destination directory of the copy. Any other `into()` should have a child spec attached and its path will be relative to the root `into()`.
+
+One final thing to be aware of is that a child copy spec inherits its destination path, include patterns, exclude patterns, copy actions, name mappings and filters from its parent. So be careful where you place your configuration.
+
+[[sec:project_copy_method]]
+==== Copying files in your own tasks
```

Comment:
I don't know. It would certainly be easier for readers to find currently if they could browse the second-level headings as well. Even if it doesn't make sense to do this in the side nav, I think having a TOC at the front of the chapter would help.

On a general note, I'm worried that we'll end up with a suboptimal structure because we're trying to satisfy a constraint of only top-level headings being visible in the navigation. It makes browsing and discovery harder than it would otherwise be.

Reply:
Hmmm...I thought I wrote something on this, but I can't remember where. I feel like the single-level heading in the navigation is forcing a structure that isn't that easy for readers to parse and comprehend. Using the copy method is still technically file copying and it would be confusing to lift it up a level just so that it appears in the side nav. And I would use the same argument for file collections and file trees, both of which should be discoverable without a page search in my view.

If I have talked about this elsewhere, sorry for repeating myself 😄

Reply:
OK, browser had a cached copy of the page without my previous comments in it. Sorry about that!